lib/eco/indentmanager.py

```diff
@@ -196,32 +189,38 @@ def apply_nodes(self, new, bol):
                 new.append(self.create_token("dedent"))
             new.append(self.create_token("newline"))
 
-            changed = self.indentation_nodes_changed(node, new)
-            if changed:
-                self.changed = True
-                # remove old indentation nodes
-                self.remove_indentation_nodes(node)
-                # and insert new ones
-                for n in new:
-                    node.insert_after(n)
-
-    def indentation_nodes_changed(self, bol, nodes):
-        """Compares indentation tokens in a line with new tokens to find out
-        whether the line needs to be updated"""
+            self.merge_nodes(node, new)
+
+    def merge_nodes(self, bol, newnodes):
         previous_nodes = []
+        last_node = bol
         node = bol.next_term
         while isinstance(node.symbol, IndentationTerminal):
             if not node.deleted:
                 previous_nodes.append(node)
             node = node.next_term
 
-        previous_nodes.reverse()
-        if len(previous_nodes) != len(nodes):
-            return True
-        for i in range(len(nodes)):
-            if nodes[i].symbol != previous_nodes[i].symbol:
-                return True
-        return False
+        for i in range(len(previous_nodes)):
+            try:
```

Comment:
Yes, the `if` version is much clearer and more obvious (not to mention faster!).